content/pubpeer.ts

```diff
@@ -39,64 +42,112 @@ function getDOI(doi, extra) {
+    const insertAfter: number = columns.findIndex(column => column.dataKey === 'title')
+    columns.splice(insertAfter + 1, 0, {
+      dataKey: 'pubpeer',
+      label: 'PubPeer',
+      flex: '1',
+      zoteroPersist: new Set(['width', 'ordinal', 'hidden', 'sortActive', 'sortDirection']),
+    })
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return columns
+  })
+
+  $patch$(itemTree.prototype, '_renderCell', original => function Zotero_ItemTree_prototype_renderCell(index, data, col) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    if (col.dataKey !== 'pubpeer') return original.apply(this, arguments)
+
+    const item = this.getRow(index).ref
+    if (!item.isRegularItem()) return ''
+
+    if (Zotero.PubPeer.ready.isPending()) { // tslint:disable-line:no-use-before-declare
+      const loading = document.createElementNS('http://www.w3.org/1999/xhtml', 'img')
+      loading.className = 'pubpeer-state-loading'
+      loading.setAttribute('src', 'chrome://zotero-pubpeer/skin/loading.gif')
+      return loading
+    }
+
+    const feedback = Zotero.PubPeer.feedback[getDOI(getField(item, 'DOI'), getField(item, 'extra'))]
+    const state = feedback.users.map(user => Zotero.PubPeer.users[user])
+
+    const cell = document.createElementNS('http://www.w3.org/1999/xhtml', 'span')
+    cell.innerText = `${feedback.total_comments}`
+    if (state.includes('priority')) {
+      cell.className = 'pubpeer-state-highlighted'
+    } else if (state.includes('neutral')) {
+      cell.className = 'pubpeer-state-neutral'
+    } else {
+      cell.className = 'pubpeer-state-muted'
+    }
+
+    return cell
+  })
+} else {
+  const itemTreeViewWaiting: Record<string, boolean> = {}
+
+  function getCellX(tree, row, col, field) {
+    if (col.id !== 'zotero-items-column-pubpeer') return ''
 
-function getCellX(tree, row, col, field) {
-  if (col.id !== 'zotero-items-column-pubpeer') return ''
+    const item = tree.getRow(row).ref
 
-  const item = tree.getRow(row).ref
+    if (item.isNote() || item.isAttachment()) return ''
 
-  if (item.isNote() || item.isAttachment()) return ''
+    if (Zotero.PubPeer.ready.isPending()) { // tslint:disable-line:no-use-before-declare
+      const id = `${field}.${item.id}`
+      if (!itemTreeViewWaiting[id]) {
+        // tslint:disable-next-line:no-use-before-declare
+        Zotero.PubPeer.ready.then(() => tree._treebox.invalidateCell(row, col))
+        itemTreeViewWaiting[id] = true
+      }
 
-  if (PubPeer.ready.isPending()) { // tslint:disable-line:no-use-before-declare
-    const id = `${field}.${item.id}`
-    if (!itemTreeViewWaiting[id]) {
-      // tslint:disable-next-line:no-use-before-declare
-      PubPeer.ready.then(() => tree._treebox.invalidateCell(row, col))
-      itemTreeViewWaiting[id] = true
+      switch (field) {
+        case 'image':
+          return 'chrome://zotero-pubpeer/skin/loading.gif'
```

Comment:
I've had problems with animated gifs in the past, ISTR Firefox will load a new thread for each individual animating gif and that ran out of threads pretty fast.